flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/DispatcherJob.java

```diff
@@ -107,18 +106,13 @@ private DispatcherJob(
 		FutureUtils.assertNoException(this.jobManagerRunnerFuture.handle((jobManagerRunner, throwable) -> {
 			// JM has been initialized, or the initialization failed
 			synchronized (lock) {
-				if (jobStatus != DispatcherJobStatus.CANCELLING) {
-					jobStatus = DispatcherJobStatus.JOB_MANAGER_RUNNER_INITIALIZED;
-				}
-
+				jobStatus = DispatcherJobStatus.JOB_MANAGER_RUNNER_INITIALIZED;
 				if (throwable == null) { // initialization succeeded
 					// Forward result future
 					jobManagerRunner.getResultFuture().whenComplete((archivedExecutionGraph, resultThrowable) -> {
-						if (archivedExecutionGraph != null) {
-							jobResultFuture.complete(DispatcherJobResult.forSuccess(archivedExecutionGraph));
-						} else {
-							jobResultFuture.completeExceptionally(ExceptionUtils.stripCompletionException(resultThrowable));
-						}
+						FutureUtils.forward(
+							jobManagerRunner.getResultFuture().thenApply(DispatcherJobResult::forSuccess),
+							jobResultFuture);
 					});
```

Comment:
This was again ill advice because by calling `thenApply` we potentially introduce a `CompletionException` which will make the `DispatcherResourceCleanupTest.testBlobServerCleanupWhenJobNotFinished` fail. Hence, I would suggest to revert this change. 

On a side note: What I meant was actually to replace the `whenComplete` call with

```
FutureUtils.forward(
	  jobManagerRunner.getResultFuture().thenApply(DispatcherJobResult::forSuccess),
	  jobResultFuture);
```